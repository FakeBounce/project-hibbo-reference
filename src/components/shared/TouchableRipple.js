import React, { PureComponent } from 'react';
import { Animated, Easing } from 'react-native';
import { number, string, func } from 'prop-types';

import { bind } from 'decko';

import TouchableRippleComponent from './components/TouchableRipple';

class TouchableRipple extends PureComponent {
  constructor(props) {
    super(props);

    this.rippleScale = new Animated.Value(0.4);
    this.rippleOpacity = new Animated.Value(0);
  }

  @bind
  onPress() {
    if (!this.props.delay) {
      this.props.onPress(this.props.value);
    }
    Animated.parallel([
      Animated.timing(this.rippleOpacity, {
        toValue: this.props.rippleMaxOpacity,
        duration: 100,
        easing: Easing.inOut(Easing.ease),
      }),
      Animated.timing(this.rippleScale, {
        toValue: 1,
        duration: 100,
        easing: Easing.inOut(Easing.ease),
      }),
    ]).start(() => {
      this.rippleOpacity.setValue(0);
      this.rippleScale.setValue(0.4);

      if (this.props.delay) {
        // Add delay to avoid freezed ripple on screen transitions
        Animated.delay(this.props.delay).start(() => {
          this.props.onPress(this.props.value);
        });
      }
    });
  }

  render() {
    return (
      <TouchableRippleComponent
        {...this.props}
        onPress={this.onPress}
        rippleScale={this.rippleScale}
        rippleOpacity={this.rippleOpacity}
      />
    );
  }
}

TouchableRipple.defaultProps = {
  value: '',
  onPress: () => {},
  delay: 10,
  rippleMaxOpacity: 0.1,
};

TouchableRipple.propTypes = {
  value: string,
  onPress: func,
  rippleMaxOpacity: number,
  delay: number,
};

export default TouchableRipple;
