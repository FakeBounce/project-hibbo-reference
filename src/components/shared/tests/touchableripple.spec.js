import { Text } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import TouchableRipple from '../components/TouchableRipple';

describe('TouchableRipple', () => {
  it('should render the component', () => {
    const text = 'test';
    const onBeginPress = () => {
      console.log('Begin Press');
    };
    const onStopPress = () => {
      console.log('Stop press');
    };
    const to = () => {
      console.log('On press');
    };

    const touchableRipple = renderer
      .create(
        <TouchableRipple onPressIn={onBeginPress} onPressOut={onStopPress} onPress={to}>
          <Text>
            {text}
          </Text>
        </TouchableRipple>,
      )
      .toJSON();

    expect(touchableRipple).toMatchSnapshot();
  });
});
