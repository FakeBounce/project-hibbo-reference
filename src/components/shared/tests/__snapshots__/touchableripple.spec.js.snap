// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchableRipple should render the component 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      test
    </Text>
  </View>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#303030",
          "borderRadius": 70,
          "height": 70,
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
          "width": 70,
        }
      }
    />
  </View>
</View>
`;
